backend/src/main/java/com/htecgroup/skynest/service/impl/CompanyServiceImpl.java

```diff
@@ -0,0 +1,42 @@
+package com.htecgroup.skynest.service.impl;
+
+import com.htecgroup.skynest.annotation.UniqueCompany;
+import com.htecgroup.skynest.exception.tier.NonExistingTierException;
+import com.htecgroup.skynest.model.entity.CompanyEntity;
+import com.htecgroup.skynest.model.entity.TierEntity;
+import com.htecgroup.skynest.model.request.CompanyAddRequest;
+import com.htecgroup.skynest.model.response.CompanyResponse;
+import com.htecgroup.skynest.repository.CompanyRepository;
+import com.htecgroup.skynest.repository.TierRepository;
+import com.htecgroup.skynest.service.CompanyService;
+import lombok.AllArgsConstructor;
+import org.modelmapper.ModelMapper;
+import org.springframework.stereotype.Service;
+import org.springframework.validation.annotation.Validated;
+
+import javax.validation.Valid;
+
+@Service
+@Validated
+@AllArgsConstructor
+public class CompanyServiceImpl implements CompanyService {
+
+  private CompanyRepository companyRepository;
+  private TierRepository tierRepository;
+  private ModelMapper modelMapper;
+
```

Comment:
@petarstoimenovskihtec  Something like this?

Reply:
what did we talk here that I need to look ?

Reply:
The custom annotation